kratos/includes/define.h

```diff
@@ -322,8 +326,11 @@ catch(...) { Block KRATOS_THROW_ERROR(std::runtime_error, "Unknown error", MoreI
 #define KRATOS_DEPRECATED
 #define KRATOS_DEPRECATED_MESSAGE(deprecated_message)
 #endif
-    
 
+#define VERSION_AS_INT (KRATOS_MAJOR_VERSION*100+KRATOS_MINOR_VERSION*10+KRATOS_PATCH_VERSION)
+#define KRATOS_DEPRECATED_VERSION(major_version, minor_version, patch_version) if (major_version*100+minor_version*10+patch_version) < VERSION_AS_INT KRATOS_DEPRECATED_MESSAGE("This method will expire eventually, please remove") else error "This method is deprecated, please remove"
+#define KRATOS_DEPRECATED_VERSION_MESSAGE(message, major_version, minor_version, patch_version) if (major_version*100+minor_version*10+patch_version) < VERSION_AS_INT KRATOS_DEPRECATED_MESSAGE(message) else error "This method is deprecated, please remove"
```

Comment:
Can we also output the message in the `#else` case? I think it could be used to give hints of what methods to use instead.

Also, `#if`, `#else` and `#error`.